### ISSUE TITLE
2.0.0~beta5: `opam pin .` does not lookup into a `./opam` directory for opam files.

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/opam#3159]
Instead it brings me into EDITOR to create an opam file.

That convention is still documented [here](https://opam.ocaml.org/doc/2.0/Packaging.html#Some-tricks) so either the doc or the implementation is wrong.

**Comment by AltGr** [comment]
It definitely should, I'll investigate. Thanks for reporting!

**Comment by AltGr** [comment]
Ah, actually, it works, but the documentation is inaccurate. The current implementation is in fact limited to:
- `foo.opam/opam` for `foo`'s metadata
- `opam/opam` (equivalent to just `opam`)

That was thought at a time where metadata generally implied multiple files, to tidy things up, but as is doesn't seem very useful (I am not sure if many, or even any, use the feature). Now with multiple packages per repo becoming so frequent, it would much more useful to have any `opam` or `*.opam` files be read like at the root of the tree when in a `opam/` (or other ? `opam.dir` ?) subdirectory.

One of the original reasons for this feature was that you can't otherwise specify a `files/` subdirectory. But why would you need it when you can just put the files in the source tree directly ?

In other words, I am open to respecifying the feature :)

**Comment by dbuenzli** [comment]
I think it would be nice if we can stuff all the opam packages in an `opam` directory, i.e. support  `opam/{$PKG1,$PKG2}.opam`.

> One of the original reasons for this feature was that you can't otherwise specify a files/ subdirectory. But why would you need it when you can just put the files in the source tree directly ?

If you really want to play nice and least surprise there's no reason why `opam/$PKG.opam/{opam,files}` even mixed with `opam/{opam,files}` can't also be supported.
